src/averageAges.js

```diff
@@ -37,7 +39,14 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = people.filter(person => (!withChildren)
+    ? person.sex === 'f'
+    : person.sex === 'f' && people
+      .find(mother => mother.mother === person.name));
+
+  const womenAge = women.map(person => person.died - person.born);
+
+  return womenAge.reduce((sum, x) => sum + x, 0) / womenAge.length;
```

Comment:
how about put it into separate function?

Reply:
what exactly?